nf_core/modules/modules_json.py

```diff
@@ -670,37 +670,53 @@ def update_subworkflow(
         if write_file:
             self.dump()
 
-    def remove_entry(self, component_type, name, repo_url, install_dir):
+    def remove_entry(self, component_type, name, repo_url, install_dir, removed_by=None):
         """
         Removes an entry from the 'modules.json' file.
 
         Args:
-            component_type (Str): Type of component [modules, subworkflows]
-            name (str): Name of the module to be removed
-            repo_url (str): URL of the repository containing the module
-            install_dir (str): Name of the directory where modules are installed
+            component_type (str): Type of component [modules, subworkflows]
+            name (str): Name of the component to be removed
+            repo_url (str): URL of the repository containing the component
+            install_dir (str): Name of the directory where components are installed
+            removed_by (str): Name of the component that wants to remove the component
         Returns:
-            (bool): True if the removal was successful, False otherwise
+            (bool): return True if the component was removed, False if it was not found or is still depended on
         """
+
+        if removed_by is None or removed_by == name:
+            removed_by = component_type
         if not self.modules_json:
             return False
         if repo_url in self.modules_json.get("repos", {}):
             repo_entry = self.modules_json["repos"][repo_url]
             if name in repo_entry[component_type].get(install_dir, {}):
-                repo_entry[component_type][install_dir].pop(name)
+                if removed_by in repo_entry[component_type][install_dir][name]["installed_by"]:
+                    repo_entry[component_type][install_dir][name]["installed_by"].remove(removed_by)
+                    if len(repo_entry[component_type][install_dir][name]["installed_by"]) == 0:
+                        self.modules_json["repos"][repo_url][component_type][install_dir].pop(name)
+                        self.dump()
+                        if len(repo_entry[component_type][install_dir]) == 0:
+
+                            self.modules_json["repos"][repo_url].pop(component_type)
+                        return True
+                    self.dump()
+                    return False
+                else:
+                    log.error(
+                        f"Could not find 'installed_by' entry for '{removed_by}' in 'modules.json' file. Did you install it first?"
```

Comment:
I was trying to remove `bam_sort_samtools` after installing `bam_sort_stats_samtools` and I got this error, which is fine, but I found it a bit confusing. Could we maybe add something like `"{name} was installed by {repo_entry[component_type][install_dir][name]["installed_by"]} and can't be removed."`